bundle/src/main/java/com/adobe/acs/commons/redirects/filter/RedirectFilter.java

```diff
@@ -0,0 +1,508 @@
+        boolean redirected = false;
+        RedirectMatch match = match(slingRequest);
+        if (match != null) {
+            RedirectRule redirectRule = match.getRule();
+            ZonedDateTime now = ZonedDateTime.now();
+            ZonedDateTime untilDateTime = redirectRule.getUntilDateTime();
+            if (untilDateTime != null && untilDateTime.isBefore(now)) {
+                log.info("redirect rule matched, but expired: {}", redirectRule.getUntilDate());
+            } else {
+                RequestPathInfo pathInfo = slingRequest.getRequestPathInfo();
+
+                String resourcePath = pathInfo.getResourcePath();
+                log.info("matched {} to {} in {} ms", resourcePath, redirectRule.toString(),
+                        System.currentTimeMillis() - t0);
+
+                String location = redirectRule.evaluate(match.getMatcher());
+                if (StringUtils.startsWith(location, "/") && !StringUtils.startsWith(location, "//")) {
+                    String ext = pathInfo.getExtension();
+                    if (ext != null && !location.endsWith(ext)) {
+                        location += "." + ext;
+                    }
+                    if (rewriteUrls()) {
+                        location = rewriteUrl(location, slingRequest.getResourceResolver());
+                    }
+                    if(preserveQueryString) {
+                        String queryString = slingRequest.getQueryString();
+                        if (queryString != null) {
+                            int idx = location.indexOf('?');
+                            if (idx == -1) idx = location.indexOf("#");
+                            if (idx != -1) location = location.substring(0, idx);
+
+                            location += "?" + queryString;
+                        }
+                    }
+
+                    if(urlAdjuster != null){
+                        location = urlAdjuster.adjust(slingRequest, location);
+                    }
+                }
+
+                log.info("Redirecting {} to {}, statusCode: {}",
+                        resourcePath, location, redirectRule.getStatusCode());
+                slingResponse.setHeader("Location", location);
+                for(String name : onDeliveryHeaders.keySet()){
+                    String value = onDeliveryHeaders.get(name);
+                    slingResponse.setHeader(name, value);
+                }
+                slingResponse.setStatus(redirectRule.getStatusCode());
+                redirected = true;
+            }
+        }
+        return redirected;
+    }
+
+    String rewriteUrl(String url, ResourceResolver resourceResolver) {
+        return resourceResolver.map(url);
+    }
+
+    @Override
+    public void destroy() {
+    }
+
+    protected boolean rewriteUrls() {
+        return rewriteUrls;
+    }
+
+    /**
+     * @return whether redirection is enabled
+     */
+    protected boolean isEnabled() {

```

Comment:
IMHO no need to map to individual fields, just expose the full configuration which has all necessary getters

Reply:
I need for mocks in junit tests.  I'd like to keep accessors to individual fields for now.